# Programming & Scripting Project

### Step 1: Importing the dataset & packages

In [1]:
# Importing the data required for analyis, using the sklearn method as demonstrated here [1]: https://www.geeksforgeeks.org/machine-learning/iris-dataset/
from sklearn.datasets import load_iris

# Importing the packages required for analysis - I used these to analyse my project for Principles of Data Analytics and feel they will be useful again in this project [2] https://www.kaggle.com/code/chats351/introduction-to-numpy-pandas-and-matplotlib:
import pandas as pd
# Numerical arrays.
import numpy as np
# Statistical data evaluation.
import matplotlib.pyplot as plt

### Step 2: Loading the Dataset & Viewing the Structure

In [ ]:
# Loading in the dataset and viewing the structure

# Load the Iris Dataset
iris = load_iris()

# Create a dataframe from the data
df = pd.DataFrame(iris.data, columns=iris.feature_names)

# Add the species column
df["species"] = iris.target

# Replace numeric species labels with names (used Co-Pilot to assist with this code [3] https://m365.cloud.microsoft/chat/conversation/65ae90ff-4d61-47e2-b093-b198808eb3a8?fromcode=cmc&redirectid=F5803B0A7CBC44BDBA083BAF6C6562EA&auth=2&internalredirect=CCM&client-request-id=bed92ccf-3ca5-4170-ba9f-eb3cc07f77a2&origindomain=CCM&login_hint=leah.curran1%40abbott.com)
df["species"] = df["species"].map({
    0: "setosa",
    1: "versicolor",
    2: "virginica"
}) 

In [3]:
# Verifying what I've done has loaded correctly
df.head()

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


In [4]:
# Viewing the data structure
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   species            150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


*df.head()* shows the first five rows of the dataset, confirming that the four variables and the species label were loaded correctly from my code above.
*df.info()* confirms that the dataset contains 150 entries and no missing values, all the columns are floats / numerical which makes them suitable for my planned analysis and plotting.

### Step 3: Summary of each Variable 

Here I'll generate summary statistics for each variable in the dataset to describe the values and spread of the data measured. The output will be saved to a txt file for reference. 

First I'll deal with the numerical variables using the describe function.

In [ ]:
# Selecting only the numerical columns
numeric_df = df.drop(columns=["species"])

# Generate the summary statistics
# Adding in rounding to make the data tidier (using this as a reference: [4] https://stackoverflow.com/questions/72862898/round-df-describe-results)
summary = numeric_df.describe().round(2)

summary

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
count,150.00,150.00,150.00,150.00
mean,5.84,3.06,3.76,1.20
std,0.83,0.44,1.77,0.76
min,4.30,2.00,1.00,0.10
25%,5.10,2.80,1.60,0.30
50%,5.80,3.00,4.35,1.30
75%,6.40,3.30,5.10,1.80
max,7.90,4.40,6.90,2.50


In [ ]:
# Creating a folder for the summary file (referencing [5] https://docs.python.org/3/library/os.html#files-and-directories)
import os 
os.makedirs("output", exist_ok=True)

In [ ]:
# Write the summary statistics to a text file (referencing [6] https://www.w3schools.com/python/python_file_write.asp)
with open ("output/summaries.txt","w") as file:
    file.write("summary-stats-iris\n\n")
    file.write(summary.to_string())